### DISCUSSION TITLE
Can I undo a "git restore"?

### DISCUSSION THREAD

**Discussion by jonatsuji** [neuefische/questions#453] [Web] [answered]
It is a short but important question.

Can I undo a git restore and how can I do it?

I hope for positive answers.
Many Thanks!

**Comment by mpagels** [comment, marked as answer]
From the official docs:

> It’s important to understand that git restore <file> is a dangerous command. Any local changes you made to that file are gone — Git just replaced that file with the last staged or committed version. Don’t ever use this command unless you absolutely know that you don’t want those unsaved local changes.

Possible solution: If you text editor is still opened with the file you have restored, then you can undo (through whatever undo command your editor supports, maybe Ctrl+Z) to get back your changes. That is - if you are lucky to have your important changes saved into memory by your editor.

**Reply by jonatsuji** [reply]
Unfortunately this doesn't work in my case, because I already switched the branch after using git restore. But it is an answer, now I know that all changes are gone. 
Thanks for your help anyway!
